apps/api/src/graphql/schema/programs.ts

```diff
@@ -66,10 +66,18 @@ input ProgramRequirementsQuery {
     programId: String!
 }
 
+input MajorIdQuery {
+    majorId: String!
+}
+
+input MinorIdQuery {
+    id: String!
+}
+
 extend type Query {
-    majors: [MajorPreview!]!
-    minors: [MinorPreview!]!
-    specializations: [SpecializationPreview!]!
+    majors(query: MajorIdQuery): [MajorPreview!]!
+    minors(query: MinorIdQuery): [MinorPreview!]!
+    specializations(query: MajorIdQuery): [SpecializationPreview!]!
```

Comment:
Even if the field is identical in name and type, it does carry a different meaning when used for the specializations query. I argue it should be a distinct type.